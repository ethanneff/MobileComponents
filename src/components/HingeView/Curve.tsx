import React, { memo } from "react";
import { Path } from "react-native-svg";
import Theme from "../../commons/Theme";

interface Props {
  curveData: string;
}

export default memo(function Curve({ curveData }: Props) {
  return (
    <Path
      d={curveData}
      fill="none"
      stroke={Theme.color.neutral300}
      strokeWidth={27}
    />
  );
});
