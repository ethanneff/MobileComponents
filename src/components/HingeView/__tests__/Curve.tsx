import { render } from "@testing-library/react-native";
import React from "react";
import Curve from "../Curve";

describe("Curve", () => {
  const curveData = `
        M 10 10 
        Q 10 15 10 20
  `;
  it("renders with the correct data", () => {
    const { container } = render(<Curve curveData={curveData} />);
    expect(container).toMatchSnapshot();
  });
});
