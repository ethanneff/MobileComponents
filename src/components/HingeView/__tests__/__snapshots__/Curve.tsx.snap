// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Curve renders with the correct data 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNSVGPath
    d="
        M 10 10 
        Q 10 15 10 20
  "
    fill={null}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "fill",
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4287603630}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth={27}
    vectorEffect={0}
    x={0}
    y={0}
  />
</View>
`;
